### DISCUSSION TITLE
How does Giraffe deal with request cancellation?

### DISCUSSION THREAD

**Discussion by xperiandri** [giraffe-fsharp/Giraffe#520] [Q&A] [answered]
How is it aligned with the article https://andrewlock.net/using-cancellationtokens-in-asp-net-core-minimal-apis/ ?

**Comment by baronfel** [comment, marked as answer]
That article talks about using HttpContext.RequestAborted as the CancellationTokenSource for your requests - this still works. If you're using `task`s, you'll need to check it yourself, and/or propagate it to your called methods.  If you're using Asyncs, you can use HttpContext.RequestAborted as the `cancellationToken` parameter of an `Async.StartAsTask` call. Either way, it mostly just works out of the box as you'd expect!

**Reply by xperiandri** [reply]
What about getting a `CancellationToken` from it and passing to async methods? How to do that easily?

**Reply by abelbraaksma** [reply]
Can you elaborate as to the scenario we're trying to solve here? Just curious. FYI, for `async`, the passing-through of the cancellation token is automatic (i.e from one nested async to the next child async etc). For `task` it must be done manually, I think that's what @baronfel was alluding to.

**Comment by ajuch** [comment]
The proposed Async.WithCancellation in https://github.com/fsharp/fslang-suggestions/issues/685 helped me solve request cancellation, maybe it helps somebody else...
